Vostok.Metrics.System/Host/NativeHostMetricsCollector_Linux.cs

```diff
@@ -0,0 +1,170 @@
+
+                cpuCollector.Collect(metrics, usedTime, systemStat.IdleTime.Value);
+            }
+
+            if (memInfo.Filled)
+            {
+                metrics.MemoryAvailable = memInfo.AvailableMemory.Value;
+                metrics.MemoryCached = memInfo.CacheMemory.Value;
+                metrics.MemoryKernel = memInfo.KernelMemory.Value;
+                metrics.MemoryTotal = memInfo.TotalMemory.Value;
+            }
+
+            if (perfInfo.Filled)
+            {
+                metrics.HandleCount = perfInfo.HandleCount.Value;
+                metrics.ThreadCount = perfInfo.ThreadCount.Value;
+                metrics.ProcessCount = perfInfo.ProcessCount.Value;
+            }
+        }
+
+        private SystemStat ReadSystemStat()
+        {
+            var result = new SystemStat();
+
+            try
+            {
+                if (FileParser.TrySplitLine(systemStatReader.ReadFirstLine(), 7, out var parts) && parts[0] == "cpu")
+                {
+                    if (ulong.TryParse(parts[1], out var utime))
+                        result.UserTime = utime;
+
+                    if (ulong.TryParse(parts[2], out var ntime))
+                        result.NicedTime = ntime;
+
+                    if (ulong.TryParse(parts[3], out var stime))
+                        result.SystemTime = stime;
+
+                    if (ulong.TryParse(parts[4], out var itime))
+                        result.IdleTime = itime;
+                }
+            }
+            catch (Exception error)
+            {
+                InternalErrorLogger.Warn(error);
+            }
+
+            return result;
+        }
+
+        private MemoryInfo ReadMemoryInfo()
+        {
+            var result = new MemoryInfo();
+
+            try
+            {
+                foreach (var line in memoryReader.ReadLines())
+                {
+                    if (FileParser.TryParseLong(line, "MemTotal", out var memTotal))
+                        result.TotalMemory = memTotal * 1024;
+
+                    if (FileParser.TryParseLong(line, "MemAvailable", out var memAvailable))
+                        result.AvailableMemory = memAvailable * 1024;
+
+                    if (FileParser.TryParseLong(line, "Cached", out var memCached))
+                        result.CacheMemory = memCached * 1024;
+
+                    if (FileParser.TryParseLong(line, "Slab", out var memKernel))
+                        result.KernelMemory = memKernel * 1024;
+                }
+            }
+            catch (Exception error)
+            {
+                InternalErrorLogger.Warn(error);
+            }
+
+            return result;
+        }
+
+        private PerformanceInfo ReadPerformanceInfo()
+        {
+            var result = new PerformanceInfo();
+
+            try
+            {
+                var processDirectories = Directory.GetDirectories("/proc/")
```

Comment:
Switch to EnumerateDirectories, get rid of ToArray to avoid unnecessary allocations.